### ISSUE TITLE
Error with nvidia-docker and the latest system76-driver-nvidia

### ISSUE THREAD

**Issue by mathisc** [NVIDIA/nvidia-docker#1131]
I just updated my computer to the latest `system-76-driver-nvidia` and I cannot run nvidia-docker anymore (this worked fine before though). 
Here is error : 
```docker run --gpus all nvidia/cuda:9.0-base nvidia-smi
docker: Error response from daemon: OCI runtime create failed: container_linux.go:346: starting container process caused "process_linux.go:449: container init caused \"process_linux.go:432: running prestart hook 0 caused \\\"error running hook: exit status 1, stdout: , stderr: nvidia-container-cli: detection error: driver error: failed to process request\\\\n\\\"\"": unknown.
ERRO[0000] error waiting for container: context canceled 
```

Here is what nvidia-smi returns on my computer : 
```+-----------------------------------------------------------------------------+
| NVIDIA-SMI 440.31       Driver Version: 440.31       CUDA Version: 10.2     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce RTX 207...  Off  | 00000000:01:00.0  On |                  N/A |
| N/A   49C    P0    32W /  N/A |    950MiB /  7982MiB |     17%      Default |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|    0      4050      G   /usr/lib/xorg/Xorg                           513MiB |
|    0      4193      G   /usr/bin/gnome-shell                         288MiB |
|    0      6110      G   ...uest-channel-token=11958984584397222172    80MiB |
+-----------------------------------------------------------------------------+
```
and here is some additional debug info : 

```
nvidia-container-cli -k -d /dev/tty info

-- WARNING, the following logs are for debugging purposes only --

I1120 23:25:39.125774 7181 nvc.c:281] initializing library context (version=1.0.6, build=0000000000000000000000000000000000000000)
I1120 23:25:39.125890 7181 nvc.c:255] using root /
I1120 23:25:39.125909 7181 nvc.c:256] using ldcache /etc/ld.so.cache
I1120 23:25:39.125926 7181 nvc.c:257] using unprivileged user 1000:1000
W1120 23:25:39.127252 7182 nvc.c:186] failed to set inheritable capabilities
W1120 23:25:39.127276 7182 nvc.c:187] skipping kernel modules load due to failure
I1120 23:25:39.127524 7183 driver.c:133] starting driver service
I1120 23:25:39.164237 7181 nvc_info.c:437] requesting driver information with ''
nvidia-container-cli: detection error: driver error: failed to process request
I1120 23:25:39.164325 7181 nvc.c:318] shutting down library context
W1120 23:25:39.174538 7181 driver.c:220] terminating driver service (forced)
I1120 23:25:39.229779 7181 driver.c:233] driver service terminated with signal 9
```

```
nvidia-container-cli -V
version: 1.0.6
build date: 2019-10-17T03:23+00:00
build revision: 0000000000000000000000000000000000000000
build compiler: x86_64-linux-gnu-gcc-7 7.4.0
build platform: x86_64
build flags: -D_GNU_SOURCE -D_FORTIFY_SOURCE=2 -Wdate-time -D_FORTIFY_SOURCE=2 -DNDEBUG -std=gnu11 -O2 -g -fdata-sections -ffunction-sections -fstack-protector -fno-strict-aliasing -fvisibility=hidden -Wall -Wextra -Wcast-align -Wpointer-arith -Wmissing-prototypes -Wnonnull -Wwrite-strings -Wlogical-op -Wformat=2 -Wmissing-format-attribute -Winit-self -Wshadow -Wstrict-prototypes -Wunreachable-code -Wconversion -Wsign-conversion -Wno-unknown-warning-option -Wno-format-extra-args -Wno-gnu-alignof-expression -I/usr/include/tirpc -g -O2 -fdebug-prefix-map=/build/libnvidia-container-Q2bRCn/libnvidia-container-1.0.6=. -fstack-protector-strong -Wformat -Werror=format-security -Wl,-zrelro -Wl,-znow -Wl,-zdefs -Wl,--gc-sections -Wl,-Bsymbolic-functions -Wl,-z,relro
```

**Comment by twilightdema** [comment]
I am experiencing the same problem.
Two of my laptops those use Pop!OS 18.04 found this error. There GPUs are GTX1050ti [Driver version 410, CUDA 10.0] and Quadro RTX5000 [Driver version 440, CUDA 10.2]..
While the another one using Ubuntu 18.04 and having GTX1070 [Driver version 410, CUDA 10.0] does not have the problem. 
I believe the root cause might be compatibility NVidia drivers from System76.

**Comment by RenaudWasTaken** [comment]
Hello!

I'm not familiar with what system76 is but it seems like a Linux distribution.

If this is the case note that neither the driver t nor the container toolkit are supported (I.e: best effort support).

That being said your error message look very much like a driver installation that failed. 

I'd suggest to look at the logs (e.g: check if the kernel modules are loaded, look at /Var/log) and reinstall the driver.

**Comment by mathisc** [comment]
Hey Renaud,

I know this issue is closed, but if that is of any help to anyone having similar issue in the future :
The driver installation seems ok and my GPU seemed to works properly locally, I only had issues when trying to attach it to a docker container. (i.e --gpus=all)
I 'solved' (or maybe more accurately 'made the issue disappear') by updating to Pop_OS 19.10 (the new release of the system 76 OS).
Thanks,
Claire
________________________________
De : Renaud Gaubert <notifications@github.com>
Envoyé : dimanche 24 novembre 2019 04:31
À : NVIDIA/nvidia-docker <nvidia-docker@noreply.github.com>
Cc : Claire Mathis <claire.mathis@hotmail.com>; Author <author@noreply.github.com>
Objet : Re: [NVIDIA/nvidia-docker] Error with nvidia-docker and the latest system76-driver-nvidia (#1131)


Closed #1131<https://github.com/NVIDIA/nvidia-docker/issues/1131>.
